### PR TITLE
Add template for Spree::Customer model proposal

### PR DESCRIPTION
**Pull request by dkubb**
This branch adds a proposal for adding a Spree::Customer model.

__NOTE__: Do not merge. This PR is for internal review and comments, after which I will close it down and then re-submit the PR against the upstream repo.

### CODE REVIEW

**Comment by dkubb** [comment]
@mbj this is ready for review.

**Comment by mbj** [comment]
@dkubb After reading it now the 5th time: No flags.

After the initial passes I had some ideas to change wording / tone etc. But after I got a full visualization about the upcoming change your RFC is not improvable by me anymore.

File: active/0000-add-customer-model.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+## Controller Changes
+
+* Replace usage of Spree::User with Spree::Customer where applicable
+* Change the admin user CRUD system to work primarily with customers
+
+The majority of changes will probably need to be performed within controllers, helpers and especially specs. Lots of specs use a user record to represent a customer in a specific context. If the proxying logic is done correctly, much of the code will continue to work, but it will still require manual verification that all instances are properly replaced. No doubt other things will be found that need touching, as is usually the case when refactoring a "god object".
+
+# Drawbacks
+
+Overall the code complexity will be reduced, but there is a risk of missing something and introducing bugs. Lots and lots of code in the specs will need to be updated.
```
Comment (mbj) [inline active/0000-add-customer-model.md]: Maybe we can mention this is also a chance to fix existing bugs, as our spec style typically exposes adjacent to change misbehavior.

Comment (dkubb) [inline active/0000-add-customer-model.md]: I think you and I know this, but I hesitate to include too much criticism in this RFC. I already mention that the user object is a god object, a fact that I doubt anyone will deny.

I am actually excited that we'll be able to rewrite some of the integration tests into finer grained unit tests.

**Comment by dkubb** [comment]
Ok, I am going to mark this as closed and open up a new PR against the upstream repo.

File: active/0000-add-customer-model.md (hunk 1 above)
Comment (mbj) [inline active/0000-add-customer-model.md]: > I already mention that the user object is a god object, a fact that I doubt anyone will deny.

Ack.

**Pull request closed without merging**